draft-sheffer-wimse-s2s-protocol.md

```diff
@@ -52,34 +52,83 @@ TODO Abstract
 
 # Introduction
 
-TODO Introduction
+This document defines authentication and authorization in the context of interaction between two workloads.
+This is the core component of the WIMSE architecture {{?I-D.ietf-wimse-arch}}.
+Assume that Service A needs to call Service B. For simplicity, this document focuses on REST services,
+and the service-to-service call consists of a single HTTP request and its response.
+We define the credentials that both services should possess and how they are used to protect the HTTP exchange.
+
+There are multiple deployment styles in use today, and they result in different security properties.
+We propose to address them differently.
+
+* Many use cases have various middleboxes inserted between pairs of services, resulting in a transport layer
+that is not end-to-end encrypted. We propose to address these use cases by protecting the REST messages at the application
+level ({{app-level}}).
+
+* The other commonly deployed architecture has a mutual-TLS connection between each pair of services. This setup
+can be addressed by a simpler solution ({{mutual-tls}}).
+
+It is an explicit goal of this protocol that a service deployment, and in fact a single call chain,
+can include both architectures. In other words, Service A can call Service B with mutual TLS protection,
+while the next call to Service C is protected at the application level.
+
+For application-level protection we currently propose two alternative solutions, one inspired by DPoP {{?RFC9449}} and
+one which is a profile of HTTP Message Signatures {{!RFC9421}}. The design team believes that we need to pick
+one of these two alternatives for standardization, once we have understood their pros and cons.
 
 ## Deployment Architecture and Message Flow
 
+Regardless of the transport between the workloads, we assume the following logical architecture:
+
 ~~~ aasvg
-+---------+
-|         |
-| This    |
-| is      |
-| a       |
-| box     |
-|         |
-+---------+
++------------+               +------------+
+|            |               |            |
+|            |               | Workload B |
+| Workload A |==============>|            |
+|            |               +------------+
+|            |               |    PEP     |
++------------+               +------------+
+      ^                            ^
+      |                            |
+      |                            |
+      v                            v
++------------+               +------------+
+|            |               |            |
+|  Identity  |               |    PDP     |
+|   Server   |               | (optional) |
+|            |               |            |
++------------+               +------------+
 ~~~
 
+The Identity Server provisions credentials to each of the workloads. At least Workload A (and possibly both) must be provisioned
```

Comment:
Reading this makes me feel like the arrow in above picture from Workload B <-> Identity Server is missing. Feel free to resolve if it's left out on purpose as it's optional.

Reply:
I omitted it for visual clarity more than anything else. I guess I should add it.

There's a related discussion that we haven't had yet on how much we want to protect the API response. If we decide to sign it then Service B absolutely needs an identity key.